RearrangingCars.java

```diff
@@ -0,0 +1,207 @@
+package CodeU_Assignment6;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class RearrangingCars {
+    /** rearrangingSteps(initial, expected) returns the steps required to arrange the cars
+     *  according to the "expected" rearrangement, starting from the "initial" rearrangement. 
+     *  
+     *  Solution:
+     *      1. Check if the current and the expected rearrangements are valid, if not, then throw an exception.
+     *      2. If the arrays are of length 0 or are exactly equal, there is no steps to be taken, hence return "".
+     *      3. Otherwise, call rearrangingStepsHelper() to get the required series of steps. 
+     *         Return the required steps as a String.           
+     * 
+     * @param initial: an array representing initial rearrangement of the cars
+     * @param expected: an array representing expected rearrangement of the cars
+     * @return String: A string containing a series of steps required to arrange the cars
+     *  according to the "expected" rearrangement, starting from the "initial" rearrangement.
+     * @throws RearrangementException 
+     */
+    public static String rearrangingSteps(int [] initial, int [] expected) throws RearrangementExceptions{
+        
+        StringBuilder str = new StringBuilder();
+        
+        //If the initial or expected rearrangement array is null, then throw an exception.
+        if(initial == null || expected == null){
+            throw new NullRearrangementsException("Either the initial rearrangement or the expected rearrangement is null");
+        }
+        
+        //If the initial and expected rearrangement array do not have same length, then throw an exception.
+        else if(initial.length != expected.length){

```

Comment:
I think there is no need for else if(), else. some if statements will do the job.